src/inversion_scripts/jacobian.py

```diff
@@ -449,173 +689,100 @@ def apply_tropomi_operator(
+            avkern_tiled = np.transpose(np.tile(avkern, (n_elements, 1)))
+            # Tile the TROPOMI dry air subcolumns
+            dry_air_subcolumns_tiled = np.transpose(
+                np.tile(dry_air_subcolumns, (n_elements, 1))
+            )  # mol m-2
+            # Derive the change in column-averaged XCH4 that TROPOMI would see over this ground cell
+            tropomi_sensitivity_gridcellIndex = np.sum(
+                avkern_tiled * sat_deltaCH4 * dry_air_subcolumns_tiled, 0
+            ) / sum(
+                dry_air_subcolumns
+            )  # mixing ratio, unitless
             # Weight by overlapping area (to be divided out later) and add to sum
-            area_weighted_virtual_tropomi += (
-                overlap_area[gridcellIndex] * virtual_tropomi_gridcellIndex
-            )  # ppb m2
-
-            # If building Jacobian matrix from GEOS-Chem perturbation simulation sensitivity data:
-            if build_jacobian:
-
-                # Get GEOS-Chem perturbation sensitivities at this lat/lon, for all vertical levels and state vector elements
-                sensi_lonlat = GEOSCHEM["Sensitivities"][iGC, jGC, :, :]
-
-                # Map the sensitivities to TROPOMI pressure levels
-                sat_deltaCH4 = remap_sensitivities(
-                    sensi_lonlat,
-                    merged["data_type"],
-                    merged["p_merge"],
-                    merged["edge_index"],
-                    merged["first_gc_edge"],
-                )  # mixing ratio, unitless
-
-                # Tile the TROPOMI averaging kernel
-                avkern_tiled = np.transpose(np.tile(avkern, (n_elements, 1)))
-
-                # Tile the TROPOMI dry air subcolumns
-                dry_air_subcolumns_tiled = np.transpose(
-                    np.tile(dry_air_subcolumns, (n_elements, 1))
-                )  # mol m-2
-
-                # Derive the change in column-averaged XCH4 that TROPOMI would see over this ground cell
-                tropomi_sensitivity_gridcellIndex = np.sum(
-                    avkern_tiled * sat_deltaCH4 * dry_air_subcolumns_tiled, 0
-                ) / sum(
-                    dry_air_subcolumns
-                )  # mixing ratio, unitless
-
-                # Weight by overlapping area (to be divided out later) and add to sum
-                area_weighted_virtual_tropomi_sensitivity += (
-                    overlap_area[gridcellIndex] * tropomi_sensitivity_gridcellIndex
-                )  # m2
+            area_weighted_virtual_tropomi_sensitivity += (
+                gridcell_dict["overlap_area"] * tropomi_sensitivity_gridcellIndex
+            )  # m2
 
         # Compute virtual TROPOMI observation as weighted mean by overlapping area
         # i.e., need to divide out area [m2] from the previous step
-        virtual_tropomi = area_weighted_virtual_tropomi / sum(overlap_area)
+        virtual_tropomi = area_weighted_virtual_tropomi / gridcell_dict["overlap_area"]
```

Comment:
I am not sure exactly whether this is still necessary... the overlap area is now the averaged overlap area...

Reply:
I don't think we need any of these area-weighting lines that multiply or divide by overlap area -- because you already dealt with the areas when building `gridcell_dicts`, right? Now you just have one average TROPOMI observation for each GEOS-Chem grid cell. So you just need to map each grid cell to a TROPOMI once (rather than doing it for each individual observation, including ones that overlap with multiple grid cells). It looks like this code is multiplying and then dividing by `gridcell_dict["overlap_area"]` in a couple of places.

I also think we should review the comments here to clarify. E.g., in places above we say "For each TROPOMI observation..." but what we're really talking about now are grid-cell-averaged TROPOMI observations.